src/v/ssx/tests/this_fiber_id_bench.cc

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "ssx/this_fiber_id.h"
+
+#include <seastar/core/reactor.hh>
+#include <seastar/core/thread.hh>
+#include <seastar/testing/perf_tests.hh>
+#include <seastar/util/later.hh>
+
+PERF_TEST(fiber_local, test_this_fiber_id) {
+    return ss::async([] {
+        auto id = ss::later()
+                    .then([] {
+                        return ss::later().then([] {
+                            return ss::later().then([] {
+                                return ss::later().then([] {
+                                    perf_tests::start_measuring_time();
+                                    auto id = ssx::this_fiber_id();
+                                    perf_tests::stop_measuring_time();
+                                    return ss::make_ready_future<uint64_t>(id);
+                                });
+                            });
+                        });
+                    })
+                    .get();
+        perf_tests::do_not_optimize(id);
+    });
+}
+PERF_TEST(fiber_local, test_this_fiber_id_deep) {
+    return ss::async([] {
+        auto id
+          = ss::later()
+              .then([] {
+                  return ss::later().then([] {
+                      return ss::later().then([] {
```

Comment:
Is there a functional difference to having this in a loop with `id = ss::later().get()`? I think that might be easier to grok what's actually being tested.

In [a newer seastar](https://github.com/scylladb/seastar/commit/b3984f892bf758960879ebc7dc69684d804f5696) `later()` is deprecated in favour of `yield()`, and the refactoring will be easier :P

Reply:
fixed that